src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,318 @@
+                    </IconButton>
+                </div>
+                <div>
+                    {this.state.texts.map((item) => {
+                        return (
+                            <div key={item.index}>
+                                <h4>
+                                    Text {item.index + 1} of{" "}
+                                    {this.state.texts.length}
+                                </h4>
+                                <p>{item.text}</p>
+                                <IconButton
+                                    onClick={() =>
+                                        this.onMoveUpClick(item.index)
+                                    }
+                                >
+                                    <KeyboardArrowUp />
+                                </IconButton>
+                                <IconButton
+                                    onClick={() =>
+                                        this.onMoveDownClick(item.index)
+                                    }
+                                >
+                                    <KeyboardArrowDown />
+                                </IconButton>
+                                <IconButton
+                                    onClick={() => this.deleteText(item.index)}
+                                    disabled={this.state.texts.length === 1}
+                                >
+                                    <DeleteForever />
+                                </IconButton>
+                            </div>
+                        );
+                    })}
+                </div>
+                <div>
+                    <Button color="primary" onClick={() => this.handleSubmit()}>
+                        Save
+                    </Button>
+                </div>
+            </div>
+        );
+    }
+}
+
+export default function ObjectEditor({
+    match: {
+        params: { sceneId, objectId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [puzzleBody, setPuzzleBody] = useState({});
+    const [puzzleType, setPuzzleType] = useState("");
+    const [animationsJson, setAnimationsJson] = useState({});
+    const [isInteractable, setIsInteractable] = useState(false);
+
+    const puzzleTypeList = [
+        { value: "text-pane", label: "Text Puzzle" },
+        { value: "rotation-controls", label: "Rotation Puzzle" },
+        { value: "keypad", label: "Keypad Puzzle" },
+        { value: "visual-pane", label: "Visual Puzzle" },
+        { value: "jigsaw-puzzle", label: "Jigsaw Puzzle" },
+        { value: "ordered-puzzle", label: "Ordered Puzzle" },
+    ];
+
+    useEffect(() => {
+        const getPuzzleBody = async () => {
+            const data = await getPuzzle(sceneId, objectId, handleError);
+            setAnimationsJson(data.animations_json);
+            if (JSON.stringify(data.animations_json) === JSON.stringify({})) {
```

Comment:
Oh interesting way of doing null check. Why do we need to do `JSON.stringify` to check if it's an empty object?

Reply:
Switched to using Object.keys(puzzleBody).length === 0 instead